src/tail_csv.c

```diff
@@ -277,6 +278,28 @@ static int tcsv_config_get_index(oconfig_item_t *ci, ssize_t *ret_index) {
   return 0;
 }
 
+static int tcsv_config_get_separator(oconfig_item_t *ci, char *ret_char) {
+  size_t len_opt;
+
+  if ((ci->values_num != 1) || (ci->values[0].type != OCONFIG_TYPE_STRING)) {
+    WARNING("tail_csv plugin: The \"%s\" config option needs exactly one "
+            "string argument.",
+            ci->key);
+    return -1;
```

Comment:
You return `-1` here, so reporting should be ERROR.

Reply:
Other invalid configuration options are also reported as `WARNING` and should then for consistency also be changed to `ERROR`, however, I see some more occurrences of `WARNING` in other plugins for ignored options, so this looks overall consistent me unless I am missing something.

Reply:
Don't look into plugins code, maintainers often do merge without deep analysis.
Look into `src/daemon/configfile.c` that maintained by "core maintainers" which are more qualified. 
The policy is simple - if function returns error, then message should be `ERROR` too, otherwise `WARNING`.

Reply:
From what I can see in the code `ERROR` is always followed by `return -1`, but there are occurrences in the code where `return -1` is preceded by `WARNING`, so I would say causality goes only in one direction here, e.g.
https://github.com/collectd/collectd/blob/6079accece816abc4ff74724d90ac499847a6246/src/daemon/configfile.c#L414-L417
However, from looking at the actual usages in the `tail_csv.c` plugin these should very well be errors and warnings should probably only allowed for backwards compatibility which is not the case here.